hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/InvocationRegistry.java

```diff
@@ -110,10 +109,10 @@ long getLastCallId() {
      */
     public boolean register(Invocation invocation) {
         final long callId;
+        boolean force = invocation.op.isUrgent() || invocation.isRetryCandidate();
         try {
-            boolean force = invocation.op.isUrgent() || invocation.isRetryCandidate();
             callId = force ? callIdSequence.forceNext() : callIdSequence.next();
-        } catch (TimeoutException e) {
+        } catch (HazelcastOverloadException e) {
```

Comment:
Can't say I'm terribly excited with this.

Reply:
Yes, I agree but it seems to be the best option for now.